src/vector_of_array.jl

```diff
@@ -61,6 +61,22 @@ function Base.append!{T, N}(VA::AbstractVectorOfArray{T, N}, new_item::AbstractV
     return VA
 end
 
+# Tools for creating similar objects
+@inline Base.similar(VA::VectorOfArray{T}) where {T} = similar(VA, T)
+@inline Base.similar(VA::VectorOfArray, ::Type{T}) where {T} = VectorOfArray([similar(VA[i], T) for i in eachindex(VA)])
```

Comment:
You can put these two together with `Base.similar(VA::VectorOfArray, ::Type{T} = eltype(VA))